### ISSUE TITLE
Support for Tuya 3.5 devices

### ISSUE THREAD

**Issue by RoboChopp** [iRayanKhan/homebridge-tuya#481]
I'm running the latest 3.1.2-beta.2 version of Homebridge-tuya and it looks like (maybe) 3.5 protocol Tuya devices aren't supported yet? I think that's what the issue is. I bought some new bulbs that connect via smart life and found the secret local key and configured the same as I have with my 3.3/3.4 version devices but it just won't discover these new ones unfortunately. Devices work find in Smart Life which is what makes me think this might be the issue.

Does this sound right? And if this is the potential reason are there plans to work on 3.5?

thankyou

**Comment by NickDarvey** [comment]
This protocol is identified by `0x00006699` instead of `0x000055aa`.

https://github.com/jasonacox/tinytuya/discussions/260
https://github.com/jasonacox/tinytuya/issues/247

**Comment by RoboChopp** [comment]
Thankyou for this @NickDarvey , is there anything I can do to implement a fix for 3.5 protocol devices?

**Comment by alexwala** [comment]
I have the same problem: recently updated my tuya lamp's, now with protocol version 3.5.

So i tried a demo homeassistant install using custom integration of [localtuya](https://community.home-assistant.io/t/local-tuya-control-tuya-devices-locally-fork-from-localtuya/634334) which supports version 3.5 and voilá the lamp's are controllable via homeassistant - so protocol  3.5 handling is already manageable 😉

**Comment by RoboChopp** [comment]
Ha nice, thanks @alexwala - I'd never tried Home Assistant before but just switched everything over to that using the local tuya setup you mentioned and rebuilt some of my custom automations using the blueprint setup. Bit of a learning curve but nothing crazy - all protocols currently available work! Thankyou

**Comment by megusd** [comment]
Same here. Got 2 new bulbs from local warehouse and they are 3.5 protocol. Had to connect through homeassistant plugin mentioned here. It works but I'm missing adaptive temperature in HomeKit.

**Comment by TheCountRobel** [comment]
Got the same issue... is there any fix out now?

**Comment by megusd** [comment]
Hi @adrianjagielak 
are you planning to add support FW 3.5 in your branch?

**Comment by adrianjagielak** [comment]
Hi @megusd 

I'm definitely open to reviewing and merging Pull Requests if anyone is able to contribute a fix, in my fork, [homebridge-tuya-plus](https://github.com/adrianjagielak/homebridge-tuya-plus).

At the moment I don't have much time to develop new features myself, but if you or someone else finds a working solution or wants to collaborate on this, feel free to [open a PR](https://github.com/adrianjagielak/homebridge-tuya-plus) or [start a discussion](https://github.com/adrianjagielak/homebridge-tuya-plus/issues/new?template=BLANK_ISSUE) in my repo, and I’ll do my best to support it.

**Comment by adrianjagielak** [comment]
...That said, I've pasted the protocol documentation to ChatGPT and it generated 3.5 protocol support for me. 😅

Not sure if it works though, I don't have 3.5 devices to test; at least it didn't break the existing support for older protocols for me (3.1-3.4).

Feel free to test it by installing **`homebridge-tuya-plus`** version **`3.3.0-beta.1`** and [checking out the PR](https://github.com/adrianjagielak/homebridge-tuya-plus/pull/13).

**Comment by PapaUte** [comment]
Hey @adrianjagielak 
I tested your plugin "homebridge-tuya-plus version 3.3.0.-beta.1

Unfortunatelly I received this error logs. 
[16/07/2025, 14:35:46] [homebridge-tuya-plus] Disconnected from Sonne_2
[16/07/2025, 14:35:10] [homebridge-tuya-plus] Socket had a problem and will reconnect to Sonne_2 (Error: ERR_PING_TIMED_OUT)
[16/07/2025, 14:35:10] [homebridge-tuya-plus] after error setting _connect in 5000ms
[16/07/2025, 14:36:06] [homebridge-tuya-plus] decrementing this._connectionAttempts, currently 2
[16/07/2025, 14:36:06] [homebridge-tuya-plus] decrementing this._connectionAttempts, currently 1
[16/07/2025, 14:38:42] [homebridge-tuya-plus] Socket had a problem and will reconnect to Sonne_2 (ECONNRESET)

Device is a simple dimmer. I am not sure, if it is Tuya 3.5 Protocol related, because I receive the same error messages with former homebridge-tuya(local/LAN) plugins with this device.

**Comment by adrianjagielak** [comment]
@PapaUte does any other local connection method (e.g. the home assistant plugin) work for you?

**Comment by PapaUte** [comment]
Local connection method works with other Tuya devices, like dimmers with protocol 3.3 and 3.4 for example. 
Also with the discovery function. 

Or are you referring to installing Home Assistant on the same raspberry, and then an home assistant plugin to homebridge?

**Comment by adrianjagielak** [comment]
@PapaUte No, I was asking if you have a way to check if this particular dimmer works with any other local open source integration (like home assistant)

**Comment by jat80** [comment]
How do you find what protocol version for the tuya device you have?

**Comment by levidhuyvetter** [comment]
@adrianjagielak have you had a chance to look more into this? Or is it still ChatGPT's code? I'm getting "Discovery v3.5 – failed to decrypt packet from X.X.X.X:6667: Unsupported state or unable to authenticate data" on all of my 3.5 bulbs

**Comment by adrianjagielak** [comment]
@levidhuyvetter still chatgpt/claude-assisted code. I don't have any devices on 3.5 to test it anyway.
I'm open to PRs though
